### PR TITLE
Set up CI and add mill-mima to check binary compatibility

### PR DESCRIPTION
**Pull request by lolgab**
Add VcsVersion to derive `publishVersion` from Git tag

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (lefou) [inline .github/workflows/actions.yml:13]: Once we use mill-vcs-version plugin, we should always use `fetch-depth: 0` with `actions/checkout`.

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Check Binary Compatibility
+        run: ./mill -i __.mimaReportBinaryIssues
+
+  publish-sonatype:
+    if: github.repository == 'com-lihaoyi/mainargs' && contains(github.ref, 'refs/tags/')
```
Comment (lefou) [inline .github/workflows/actions.yml:32]: `s/contains/startsWith/` is probably more accurate.

File: build.sc [hunk 3]
```diff
@@ -15,8 +18,9 @@ val scalaNativeVersions = for {
   scalaNativeV <- Seq("0.4.0")
 } yield (scalaV, scalaNativeV)
 
-trait MainArgsPublishModule extends PublishModule with CrossScalaModule {
-  def publishVersion = "0.2.1"
+trait MainArgsPublishModule extends PublishModule with CrossScalaModule with Mima {
+  def publishVersion = VcsVersion.vcsState().format()
+  def mimaPreviousVersions = VcsVersion.vcsState().lastTag.toSeq
```
Comment (lefou) [inline build.sc:23]: This would silently accept a missing last tag, whereas we always want to use the last tag.

**Comment by lolgab** [comment]
Thanks for the review @lefou . I address the problems and once approved I port the changes to the other repositories.

**Comment by lolgab** [comment]
@lefou Can I ask you for another review round?

**Review by lefou — APPROVED** [review]
Looks good to me. (Sorry for the late reply, I was offline for while on a bike tour.)

**Comment by lolgab** [comment]
No worries :-)
Thank you!

**Pull request merged**